code/modules/mob/living/abilities/step_strike.dm

```diff
@@ -77,6 +77,9 @@
 
 		target_loc = pick(candidates)
 
+		if(!check_trajectory(target_loc, user, user.pass_flags))
```

Comment:
if the target location isnt reachable then pick_n_take another, don't just randomly fail